File: week6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text:")
letters = 0
words = 1
sentences = 0
n = len(text)

text = text.lower()

for c in text:

    # count letters

    if (64 < ord(c) < 91) or (96 < ord(c) < 123):
        letters = letters + 1

    # count words

    elif c == " ":
        words = words + 1

    # count sentences

    elif c == "." or c == "!" or c == "?":
        sentences = sentences + 1

# calculate index

L = (letters / words) * 100
S = (sentences / words) * 100
index = 0.0588 * L - 0.296 * S - 15.8

grade = round(index)

# print(grade)
if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{

    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;
    int n = strlen(text);

    for (int i = 0; i < n ; i++)
    {
        //Capital letters become small letters
        if (isupper(text[i]))
        {
            text[i] = tolower(text[i]);
        }
        //count letters
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            letters = letters + 1;
        }
        //count words
        else if (text[i] == ' ')
        {
            words = words + 1;
        }
        //count sentences
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences = sentences + 1;
        }
    }
    //calculate index
    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;
    float index = 0.0588 *  L -  0.296 *  S - 15.8;
    //round the index
    int grade = round(index);

    printf("L: %f\n", L);
    printf("S: %f\n", S);

    //print grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}